xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/renderer/xhtml/XHTMLChainingRenderer.java

```diff
@@ -591,7 +586,7 @@ public void onImage(ResourceReference reference, boolean freestanding, Map<Strin
     public void onRawText(String text, Syntax syntax)
     {
         // Directly inject the HTML content in the wiki printer (bypassing the XHTML printer)
```

Comment:
You also need to change this comment IMO (since it's not HTML). The comment should also mention the fact that reason we directly inject these is  because this output is primarily meant for browsers. If this renderer is to be used for something else than browsers, I don't know if it would work to inject all of them directly (e.g. this is supposed to be a XHTMLChainingRendered, thus it's supposed to produce XHTML only. However we inject HTML (and that's NOT XHTML). So if the output is read using an XML parser for example this will fail!! 

TBH I'm not sure that it's correct that an XHTMLChainingRenderer directly outputs HTML. It should first convert it to XHTML IMO. We might have a bug here. And if it were done like this, then the code would need to check for XHTML-type syntaxes and not BROWSER_TYPES...

We need to check who did this change and verify with him/her about this point IMO.